### DISCUSSION TITLE
Aliasing files in service worker context

### DISCUSSION THREAD

**Discussion by evanminto** [parcel-bundler/parcel#7470] [Q&A]
I’m trying to set up “service worker rendering” wherein the service worker renders the “server-side” code, going beyond just caching. Just like isomorphic rendering, doing so requires that I alias certain files for different contexts. Parcel makes it easy to do these aliases for the main two contexts: Node and the browser. But neither context fits the requirements of the service worker. For example:

* The service worker doesn't have access to window or DOM APIs.
* The service worker doesn't have access to Node modules or built-in Node packages.

I need a way to select aliased versions of packages that don't need DOM APIs without giving up the browser context behavior of bundling Node modules.

Is there a way to do this that I’m missing? A plugin maybe?

**Comment by mischnic** [comment]
There is no standardized way to do this (like`package.json#browser`), but a resolver plugin could read `dependency.env.context` and resolve differently based on that.
